delphix/virtualization-sdk
# DLPX-79403 ENVIRONMENT_DISCOVER operation fails with PluginSocketNotReadyException for EBS vsdk plugin on dvp 4.0.2

# Context:
When plugin developers add symlink folders to /src directory, `dvp build` does not create .pyc files in the same folder because of the missing `legacy=true` when compiling build folder.

Plugin developers can use symlink folders as they have common code which they would like to use when developing plugins for different systems like WINDOWS or UNIX.

# Problem:
Absence of .pyc files for symlinks folder after build fails the plugin imports.

# Solution:
When we copy the src folder to build folder, we copy the symlink folder as a folder to build folder. Compiling the build folder after the copy using `legacy=true` creates the .pyc files in the same folder which can be copied to zip files.

# Testing
- Build and upload a plugin with dvp==4.0.2 having symlink folder and symlink files in /src folder and /src/test1 folder (for hierarchy check). It failed because of import error.
- Build and upload the same plugin with the changes and worked.

# Notes To Reviewers:
- This change does not need code changes to delphix engine related to dvp-api.
